torchmdnet/models/utils.py

```diff
@@ -77,6 +79,164 @@ def message(self, x_j, W):
         return x_j * W
 
 
+class OptimizedDistance(torch.nn.Module):
```

Comment:
The reason I did not did this is because I am weary of a current use of Distance relying on something unexpected (maybe related to the ordering?). I am waiting for regular users of each model to tell me "I trained with the new Distance and everything is ok". Maybe a bit superstitious, but if it is all the same I would rather do another PR to replace current uses.